core/src/services/dropbox/core.rs

```diff
@@ -0,0 +1,195 @@
+            path: path.to_string(),
+        };
+        let request = self
+            .header_params(self.sign(Request::post(&url)), &serde_json::to_string(&download_args).unwrap())
+            .body(AsyncBody::Empty)
+            .map_err(new_request_build_error)?;
+        self.client.send(request).await
+    }
+
+    pub async fn dropbox_update(
+        &self,
+        path: &str,
+        size: Option<usize>,
+        content_type: Option<&str>,
+        body: AsyncBody,
+    ) -> Result<Response<IncomingAsyncBody>> {
+        let url = "https://content.dropboxapi.com/2/files/upload".to_string();
+        let args = DropboxUploadArgs {
+            path: path.to_string(),
+            mode: "overwrite".to_string(),
+            mute: true,
+            autorename: false,
+            strict_conflict: false,
+        };
+        let mut request_builder = Request::post(&url);
+        if let Some(size) = size {
+            request_builder = request_builder.header(header::CONTENT_LENGTH, size);
+        }
+        if let Some(mime) = content_type {
+            request_builder = request_builder.header(header::CONTENT_TYPE, mime);
+        }
+        let request = self
+            .header_params(self.sign(request_builder), &serde_json::to_string(&args).unwrap())
+            .body(body)
+            .map_err(new_request_build_error)?;
+
+        self.client.send(request).await
+    }
+
+    pub async fn dropbox_delete(&self, path: &str) -> Result<Response<IncomingAsyncBody>> {
+        let url = "https://api.dropboxapi.com/2/files/delete_v2".to_string();
+        let args = DropboxDeleteArgs {
+            path: path.to_string(),
+        };
+        let request = self
+            .sign(Request::post(&url))
+            .header(header::CONTENT_TYPE, "application/json")
+            .body(AsyncBody::Bytes(Bytes::from(serde_json::to_string(&args).unwrap())))
+            .map_err(new_request_build_error)?;
+        self.client.send(request).await
+    }
+
+    pub async fn dropbox_copy(&self, from: &str, to: &str) -> Result<Response<IncomingAsyncBody>> {
+        let url = "https://api.dropboxapi.com/2/files/copy_v2".to_string();
+        let args = DropboxCopyArgs {
+            from_path: from.to_string(),
+            to_path: to.to_string(),
+            allow_ownership_transfer: false,
+            allow_shared_folder: false,
+            autorename: false,
+        };
+        let request = self
+            .sign(Request::post(&url))
+            .header(header::CONTENT_TYPE, "application/json")
+            .body(AsyncBody::Bytes(Bytes::from(serde_json::to_string(&args).unwrap())))
+            .map_err(new_request_build_error)?;
+        self.client.send(request).await
+    }
+
+    pub async fn dropbox_move(&self, from: &str, to: &str) -> Result<Response<IncomingAsyncBody>> {
+        let url = "https://api.dropboxapi.com/2/files/move_v2".to_string();
+        let args = DropboxRenameArgs {
```

Comment:
We can implement `Default` for `DropboxRenameArgs` so that we can:

```
let args = DropboxRenameArgs {
    from_path: from.to_string(),
    to_path: to.to_string(),
    ..Default::default(),
};
```